packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -1,20 +1,123 @@
-import { BaseLoader, PageResourceStatus } from "./loader"
+import {
+  BaseLoader,
+  PageResourceStatus,
+  getStaticQueryResults,
+  getPageDataDb,
+  getPageDb,
+} from "./loader"
 import { findPath } from "./find-path"
+import getSocket from "./socketIo"
+import normalizePagePath from "./normalize-page-path"
+
+function didDataChange(a, b) {
+  return JSON.stringify(a) !== JSON.stringify(b)
+}
```

Comment:
From previous playing with socket.io & xhr the data always changed because of a different order of keys. If it's not the case for you, you can ignore this comment :smile:

So I would suggest moving to isEqual from lodash as it's already is a dependency.

```suggestion
// TODO move away from lodash
import isEqual from "lodash/isEqual"

function didDataChange(a, b) {
  return !isEqual(a, b)
}
```